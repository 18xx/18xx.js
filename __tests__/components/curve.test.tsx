import * as React from 'react';
import { ReactElement } from 'react';
import * as renderer from 'react-test-renderer';

import Curve from '../../src/components/curve';

import Point from '../../src/point';

describe('Curve', () => {
  describe('#toString', () => {
    it('returns an svg path element', () => {
      const subject: ReactElement<Curve> = (
        <Curve
          point1={new Point(1, 2)}
          point2={new Point(3, 4)}
          radius={5}
        />
      );
      expect(renderer.create(subject)).toMatchSnapshot();
    });
  });
});
