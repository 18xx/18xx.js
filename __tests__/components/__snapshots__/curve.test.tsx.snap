exports[`Curve #toString returns an svg path element 1`] = `
<path
  d="M1 2 A 5 5 0 0,0 3 4"
  fill="none"
  stroke="black"
  strokeWidth="8" />
`;
